File: conformance/bundles/02-context-isolation/CounterSteps.java
package dev.varar.conformance.bundle02;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;

/** Java sibling of {@code counter.steps.ts} / {@code counter.steps.py} (bundle {@code 02-context-isolation}). */
public final class CounterSteps implements StepDefinitions<CounterSteps.Ctx> {

    record Ctx(int count) implements State {}

    @Override
    public void register(Steps<Ctx> s) {
        s.defineState(() -> new Ctx(0));

        s.stimulus("I increment", (Ctx ctx) -> new Ctx(ctx.count() + 1));

        s.sensor("The count is {int}", (Ctx ctx, Integer n) -> ctx.count());
    }
}

File: conformance/bundles/02-context-isolation/counter.steps.kt
@file:JvmName("CounterSteps")

// Kotlin sibling of counter.steps.ts / counter.steps.py / CounterSteps.java
// (bundle 02-context-isolation).
package dev.varar.kotlin.conformance.bundle02

import dev.varar.kotlin.stimulus
import dev.varar.kotlin.steps
import dev.varar.kotlin.sensor

data class Ctx(val count: Int = 0)

val steps = steps(::Ctx) {
    stimulus("I increment") { copy(count = count + 1) }
    sensor("The count is {int}") { n: Int -> count }
}
